From e212e3f63ae499b1da079f2ac7b74d5506828c76 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Thu, 5 Feb 2015 16:38:52 +0800
Subject: [PATCH] MLK-10218-1 charger: imx6_usb: change detection timeout to
 500ms

With IC engineer confirmation, 500ms is enough to detect whether
it is a standard USB connection or not. Decrease the timeout value
can reduce the system suspending time if it is not a standard
USB connection.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 drivers/power/imx6_usb_charger.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/imx6_usb_charger.c b/drivers/power/imx6_usb_charger.c
index 7ec8ace..1587c6a 100644
--- a/drivers/power/imx6_usb_charger.c
+++ b/drivers/power/imx6_usb_charger.c
@@ -107,7 +107,7 @@ static int imx6_usb_charger_detect(struct usb_charger *charger)
 				usleep_range(5000, 10000);
 		} else {
 			data_pin_contact_count = 0;
-			msleep(20);
+			usleep_range(5000, 6000);
 		}
 	}
 
-- 
1.8.0

